### PR TITLE
#28085

### PR DESCRIPTION
**Pull request by eidng8**
### What does this PR do?

Trying to make a PR that was provided by @adithep back in 2015.
### What issues does this PR fix or reference?

This adds an extra minion configuration on Mac OS:

```
docker.machine: 'your machine name'
```
### New Behavior

The new configuration designates on which docker machine to operate.
### Tests written?

No

First time making a PR, hope it gets well...

### CODE REVIEW

**Comment by eidng8** [comment]
Don't have a clue on the error. Build log shows "No changes. Changes in dependency", "no merge conflicts", & "No identified problem".

Any pointer?

**Comment by eidng8** [comment]
Regarding tests, how should I make a test against the PR?

**Comment by eidng8** [comment]
hm...

I'v see

```
04:55:02 [xUnit] [INFO] - [JUnit] - No test report file(s) were found with the pattern 'artifacts/unittests/*.xml' relative to '/var/jenkins/workspace/PR/salt-pr-rs-cent7-n' for the testing framework 'JUnit'.  Did you enter a pattern relative to the correct directory?  Did you generate the result report(s) for 'JUnit'?
04:55:02 [xUnit] [ERROR] - No test reports found for the metric 'JUnit' with the resolved pattern 'artifacts/unittests/*.xml'. Configuration error?.
04:55:02 [xUnit] [INFO] - Failing BUILD.
```

in both ubuntu14 & cent7 build logs.

Is it my fault?

**Comment by rallytime** [comment]
@eidng8 We had some problems with our test suite, which we're resolving. I'll just trigger them to re-run. Thanks!

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by eidng8** [comment]
phew... (swiping sweat)

**Comment by eidng8** [comment]
Following up the change, I think there are something to be done further.

Currently the config is hard-coded to minion config. It is not suitable, as in my case, there are multiple docker machines running on the Mac box. After looked a bit into the code, it seems that adding a `machine` parameter to state/module files would require changes to many (if not most) of `dockerng` functions. Though every change would be just a few lines to pass the argument along.

I put this here for later reference, in case I forget in the future.

Also, would like to submit a new issue after this PR gets merged.

**Comment by cachedout** [comment]
@ticosax or @terminalmage Any comments here?

File: salt/modules/dockerng.py [hunk 1]
```diff
@@ -755,7 +755,26 @@ def _get_client(timeout=None):
...
+                client_cert=(docker_machine_tls['ClientCertPath'],
+                             docker_machine_tls['ClientKeyPath']),
+                ca_cert=docker_machine_tls['CaCertPath'],
+                assert_hostname=False,
+                verify=True)
+        except Exception as exc:
+            raise CommandExecutionError(
+                'Docker machine {0} failed: {1}'.format(docker_machine, exc))
+
+    __context__['docker.client'] = docker.Client(**client_kwargs)
```
Comment (terminalmage) [inline salt/modules/dockerng.py]: Why is the indentation level for this line changed?

**Comment by eidng8** [comment]
um... 

Sorry, but I can't see which line you've commented on...

**Comment by cachedout** [comment]
@eidng8 He is comparing L758 to L777.

**Pull request merged**